spring-data-vmware-gemfire/src/main/java/org/springframework/data/gemfire/repository/query/StringBasedGemfireRepositoryQuery.java

```diff
@@ -334,6 +334,11 @@ else if (isSingleNonEntityResult(queryMethod, collection)) {
 			return collection.iterator().next();
 		}
 		else {
+			getLogger().error(String.format("""
+              Query: %s
+              Query method return type: %s
+              Returned results: %s""",
```

Comment:
not sure that printing out the results of here... possibly think more types expected vs provided not printout the query result

Reply:
I could do that. But by the time the exception is thrown, we've already successfully turned the result into a collection. The real info we want in this case is the number of things returned and what they are, more than the type of result gemFire wrapped it in.

Reply:
Changed